src/lib.rs

```diff
@@ -0,0 +1,289 @@
+// Portions Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+
+//! Provides a wrapper for allocating, handling and interacting with the guest memory regions.
```

Comment:
Can you add some examples of how the vm-memory is going to be used?

Reply:
add example to lib.rs?
I have added an simple example to README.md.

Reply:
The benefit of having examples in the code is that they're also run with cargo test and published along with the public interface and the documentation to [docs.rs](https://docs.rs/) whenever you run `cargo publish`. Also, it is a nice way to keep examples up to date with the code (cargo test will crash if the examples are out of date). From my point of view it is easier to start working with a crate if you have examples of how to use the public interface in the docs.

We can have this is as a nice to have feature. What are your thoughts on this?

Reply:
Good suggestion. 
How about covering it with a coming PR? We should also ensure the content generated by "cargo doc" 
is suitable.